kong/db/dao/targets.lua

```diff
@@ -250,9 +250,22 @@ function _TARGETS:page_for_upstream_with_health(upstream_pk, ...)
     -- Note that lua-resty-dns-client does retry by itself,
     -- meaning that if DNS is down and it eventually resumes working, the
     -- library will issue the callback and the target will change state.
-    target.health = health_info
-                   and (health_info[target.target] or "DNS_ERROR")
-                   or  "HEALTHCHECKS_OFF"
```

Comment:
What happened to the HEALTHCHECKS_OFF case in this test? Does `get_upstream_health` always return a `health_info` in line 238 above now? From my reading below it seems that it would end up defaulting to `UNHEALTHY`...?

I'm not saying this is broken because we do have a test in the suite that verifies that HEALTHCHECKS_OFF is still returned and that is passing, but I'm confused and I'd like to understand the code changes a bit better.

Reply:
Thanks, it was really broken. I added a test for that and fixed the test.